XenAdmin/Wizards/PatchingWizard/PatchingWizard_UploadPage.cs

```diff
@@ -509,6 +509,12 @@ private void singleAction_Completed(ActionBase sender)
                                 AllIntroducedPoolUpdates.Add(PoolUpdate, SelectedNewPatchPath);
                             }
                         }
+
+                        foreach( var dict in (action as UploadSupplementalPackAction).SrUploadedUpdates)
+                        {
+                            if (!SrUploadedUpdates.ContainsKey(dict.Key))

```

Comment:
We should replace the values in the dictionary even if the key exists already, because it could mean that we uploaded a different update to a different SR. 
For example: I upload update U1 to SR S1; then I detach the SR; the precheck fails, I go back and select a different update, U2, which will be uploaded to a different SR, say S2 (if S1 is still detached). In this case I want S2 to be saved here, not S1. But then what happens if I go back and select update U1 again?

Reply:
Thanks.I think you're right and replace the values in the dictionary. 
Because I think we can uploaded just one updates for a host once(except the Automated Updates because its logic is different here ). So in the case you mentioned above, if U1 can be uploaded again, then S2 can be updated to S1. If U1 cannot be uploaded again, then S2 shouldn't be updated.

Reply:
Actually U1 will not be uploaded again, because the wizard knows that it was uploaded already (to S1), and the wizard will allow the user to continue to the precheck step with the SrUploadedUpdates containing S2, which is not correct (as S2 doesn't contain U1). The precheck step will not flag the SR as broken., because it doesn't check S1, it checks S2.